Functions/InputOutput/xASL_io_ReadDataPar.m

```diff
@@ -1,12 +1,12 @@
-function [x] = xASL_io_ReadDataPar(pathDataPar,bStudyPar)
+function [x] = xASL_io_ReadDataPar(pathDataPar, bStudyPar)
```

Comment:
We had 10 functions for this that I have merged to one on a previous request. We can split, but please create a new issue for that.

Reply:
I cannot imagine that I would have approved merging 10 different fruits, and then calling them all "apple" ;)
Can you please write the issue? You know more details of what happened last time, and why they are joint.

To me, it doesn't seem sensible to join them. Reading sourceStructure (for dcm2nii), reading studyPar (for nii2BIDS), reading dataPar (for processing) should all be different, if we generalize this we loose modularity and things could go wrong? So I don't see the benefit of why you mixed this in the first place?

Perhaps I can understand historically that we read the dataPar for the parameters that belong to the data, that now are in the JSON sidecars, and `derivatives` and `temp` both have the legacy structure.

This is a remnant from a merged import and processing, we should separate this ;)

Reply:
We had 2-3 functions for reading JSON with bizarre names. I have merged them to a single one (historically called ReadDataPar), but we can rename it to readJSON or something else.

It's main purpose is to read and fix manually created JSONs = studyPar and dataPar. It's not used for anything else. No sourcestructure json.

The function is exactly the same for studyPar and dataPar and the only difference is that for dataPar, we correct for any errors in data-parameters - which is not needed for studyPar, because there shouldnt' be any dataPars.

That's the obvious benefit of having a single function. And makes no sense to have two functions. If you want, we can remove this option and move this parameter correction outside - I can then create an issue for that or even do it here - up to you.

But it is wrong to regard this as merged import and processing - this has nothing to do with it, it's just a service function.

Reply:
OK, I created a new issue for this: #1193